internal/backuprestore/backup.go

```diff
@@ -35,185 +34,125 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/wait"
-	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/yaml"
 )
 
-// ReconcileBackup reconciles the backup CRs
-func (h *BRHandler) ReconcileBackup(ctx context.Context, oadpContent []lcav1alpha1.ConfigMapRef,
-) (
-	result ctrl.Result, status BackupStatus, err error,
-) {
-	if len(oadpContent) == 0 {
-		status.Status = BackupFailedValidation
-		status.Message = "No oadp configmap is provided."
-		result.RequeueAfter = 1 * time.Minute
-		return
-	}
+type BackupTracker struct {
+	PendingBackups          []string
+	ProgressingBackups      []string
+	SucceededBackups        []string
+	FailedBackups           []string
+	FailedValidationBackups []string
+}
 
-	oadpConfigmaps, err := getConfigMaps(ctx, h.Client, oadpContent)
-	if err != nil {
-		if !k8serrors.IsNotFound(err) {
-			return
-		}
+// BackupPhase defines the phase of backup
+type BackupPhase string
+
+// Constants for backup phase
+const (
+	BackupPending          BackupPhase = "BackupPending"
+	BackupFailedValidation BackupPhase = "BackupFailedValidation"
+	BackupFailed           BackupPhase = "BackupFailed"
+	BackupCompleted        BackupPhase = "BackupCompleted"
+	BackupInProgress       BackupPhase = "BackupInProgress"
+	BackupSkipped          BackupPhase = "BackupSkipped"
+	BackupRequested        BackupPhase = "BackupRequested"
+)
```

Comment:
As we are not returning a status, could these be removed?

Reply:
Removed the custom status and instead using velvero status.phase